src/borg/archive.py

```diff
@@ -1115,6 +1116,12 @@ def _open_rb(path):
                 raise
             # Was this EPERM due to the O_NOATIME flag? Try again without it:
             return os.open(path, flags_normal)
+        except OSError as exc:
+            # O_NOATIME causes EROFS when accessing a volume shadow copy in WSL1
+            if 'retryreadonly' in workarounds and exc.errno == errno.EROFS:
+                return os.open(path, flags_normal)
+            else:
+                raise exc
```

Comment:
guess this is just `raise`, see line 1116.

also, check the code in 1114..1116 - if we do not have NOATIME at all the initial open did not use it, so there is no need to try it again with same flags.